#163544772: endpoint to create a political party

Bobjayafam opened the pull request:
#### What does this PR do?
create an endpoint ``POST /parties`` for creating political parties
#### Description of Task to be completed?
- create error handling middleware
- create parties model
- write middleware for validating inputs
- write middleware for handling image uploads
- add a party controller class and a create party method for the route
- write tests for the endpoint
#### How should this be manually tested?
- clone the GitHub repo
- checkout branch ``ft-admin-create-party-163544772``
- on POSTMAN, send a POST request to ``/api/v1/parties``. with form data selected, send the request body to contain name, hqAddress, logoUrl, acronym.logoUrl should be of file type. The response should be the created party with a status code of 201
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/story/show/163544772
#### Screenshots (if appropriate)
![create-party-api](https://user-images.githubusercontent.com/19153324/51878935-ca128a00-2371-11e9-84dd-67b925dc9e29.png)

Bobjayafam committed:
feat(server): endpoint to create a political party

- create error handling middleware
- create parties model
- write middleware for validating inputs
- write middleware for handling image uploads
- add a party controller class and a create party method for the route

[Finishes #163544772]

Bobjayafam commented:
@akinwale-habibullah kindly review. Thanks

akinwale-habibullah reviewed (commented):
![image](https://user-images.githubusercontent.com/24890672/51970131-5dcd7e80-2476-11e9-91ac-09ca710eb18f.png)

Kudos for taking initiative to save the uploaded file to cloudinary. However, here's the response I get when I make a POST request to the /parties endpoint. You might want to revisit this.

Bobjayafam committed:
feat(server): add mkdirp for creating the image upload folder

[Finishes #163544772]

Bobjayafam commented:
@akinwale-habibullah I have rectified the uploading error with the latest commit.

akinwale-habibullah commented on server/middleware/imageUpload.js:
```diff
@@ -0,0 +1,73 @@
+
+dotenv.config();
+
+const cloudinary = require('cloudinary').v2;
```
Why are you mixing require and import. Please stick with import.

Bobjayafam replied:
Ok

akinwale-habibullah commented on server/middleware/imageUpload.js:
```diff
@@ -0,0 +1,73 @@
+let logoUrlUploaded;
+
+cloudinary.config({
+  cloud_name: process.env.cloud_name,
```
Please specify these environment variables in the README file, under the installation section, so a user of your repo knows they're required to supply these variables.

akinwale-habibullah commented on server/controllers/PartyController.js:
```diff
@@ -0,0 +1,32 @@
+class PartyController {
+  static createParty(req, res, next) {
+    const {
+      name, hqAddress, logoUrl, acronym,
```
logoUrl variable is unused in this module. Any idea why?

Bobjayafam replied:
@akinwale-habibullah I forgot to remove it. It's unused because it is parsed as req.file instead of req.body.

akinwale-habibullah replied:
Ok. Great you've removed it now.

akinwale-habibullah reviewed (commented):
Please see my comments. Add the changes.

Kudos for the good job so far

Bobjayafam committed:
feat(server): implement feedback

- use import instead of require
- add sample .env
- update README.md

[Finishes #163544772]

Bobjayafam commented:
Thank you
